src/KioskMode/README.md

```diff
@@ -0,0 +1,102 @@
+# Kiosk-Mode
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
+interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+**Table of Contents**
+* [Purpose](#purpose)
+* [Architecture](#architecture)
+* [Implementing a Provider](#implementing-a-provider)
+* [Implementing a Player](#implementing-a-player)
+
+## Purpose
+
+A kiosk mode is understood to be a special mode to display certain applications, or
+in the context of ILIAS, certain views in an application where the user gets less
+control over the application or feature than usual. This is traditionally used for
+the unsupervised operation of the application at public locations like museums or
+train stations, where the user e.g. should not be able to perform otherwise common
+operations like closing a window or shutting down the system.
+
+Similar requirements arise in ILIAS for different objects and different scenarios:
+
+* A Test needs to be displayed in a restricted way for the use in an E-Exam.
+* The LTI-Interface needs the ability to display objects in a Kiosk-Mode when it
+  ILIAS is used as a Tool-Provider.
+* The Learning Sequence requires a kiosk mode to maintain control over the general
+  display and navigation.
+
+This library facilitates the implementation of objects that can be viewed in a kiosk
+mode and of features that display these objects in a kiosk mode. The provided means
+aim to make it possible for objects to implement a kiosk mode only once and then be
+used by all features that need to display objects in that kiosk mode. On the other
+hand, features that require viewing other objects in a kiosk mode should be able to
+use all objects that implement a kiosk mode.
```

Comment:
The existing LTI mode does not make any restrictions regarding the offered features of an object (it is possible to offer administrative screens). Users might have access to all screens of the object (e.g. learners might be allowed to edit learning modules). If we want to keep this feature set it would mean to rewrite more or less all screens of the object. Is this really the way to proceed?